lib/pure/collections/sequtils.nim

```diff
@@ -487,6 +487,27 @@ proc keepIf*[T](s: var seq[T], pred: proc(x: T): bool {.closure.})
       inc(pos)
   setLen(s, pos)
 
+proc countIt*[T](s: openArray[T], pred: proc(x: T): bool {.closure.}): int {.inline, since: (1, 2).} =
+  ## Returns a count of all the items that fulfilled the predicate.
+  ##
+  ## Note that `s` must be declared as a ``var``.
+  ##
+  ## See also:
+  ## * `count proc<#count,openArray[T],T>`_
+  runnableExamples:
+    let
+      numbers = @[1, 2, 3]
+      f1 = countIt(numbers, proc(x: int): bool = x < 3)
```

Comment:
nit: u can remove 3 lines but avoid definiing f1 etc, eg:
`assert countIt(numbers, proc(x: int): bool = x < 3) == 2`
which is also more idiomatic/easier to read